utils/managed_bytes.hh

```diff
@@ -458,10 +458,20 @@ public:
     managed_bytes_fragment_range_basic_view<mutable_view::yes> as_fragment_range() noexcept;
     managed_bytes_fragment_range_basic_view<mutable_view::no> as_fragment_range() const noexcept;
 
-    iterator begin(size_t offset = 0);
+    iterator begin() {
+        if (external()) {
+            return iterator(_u.ptr->data, _u.ptr->data + _u.ptr->frag_size, _u.ptr->next);
+        }
+        return iterator(_u.small.data, _u.small.data + _u.small.size, nullptr);
+    }
     iterator end() { return iterator(); }
 
-    const_iterator begin(size_t offset = 0) const;
+    const_iterator begin() const {
+        if (external()) {
+            return const_iterator(_u.ptr->data, _u.ptr->data + _u.ptr->frag_size, _u.ptr->next);
+        }
+        return const_iterator(_u.small.data, _u.small.data + _u.small.size, nullptr);
+    }
     const_iterator end() const { return const_iterator(); }
```

Comment:
begin/end are dangerous, since they are slow and people might use them acceidentally.

Perhaps we can have something like

```c++
    template <typename Func>
    requires std::invocable<Func, managed_bytes::iterator, managed_bytes::iterator> 
          && std::invocable<Func, bytes_view::iterator, bytes_view::iterator> 
    auto managed_bytes::with_iterator_pair(Func func) {
         if (is_fragmented()) {
             return func(slow_begin(), slow_end());
         } else {
             return func(single_fragmented_begin(), single_fragment_end());
         }
    }
```